cmd/container: running with `no_shim = true` crashes containerd

Running with [`no_shim = true`](`https://github.com/containerd/containerd/blob/62918511f32014927e20342dd3e29c8a72fda25b/linux/runtime.go#L46`) will cause the following panic when calling `ctr list`:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x50 pc=0xc50c13]

goroutine 110 [running]:
github.com/containerd/containerd/linux/shim.(*initProcess).ContainerStatus(0x0, 0x1363e00, 0xc4204706c0, 0x1363e00, 0xc4204706c0, 0xe31506, 0x4)
	/home/sjd/go/src/github.com/containerd/containerd/linux/shim/init.go:128 +0x83
github.com/containerd/containerd/linux/shim.(*Service).State(0xc42011c420, 0x7f3d7b2a4248, 0xc4204706c0, 0x13aaf90, 0x0, 0x0, 0x0)
	/home/sjd/go/src/github.com/containerd/containerd/linux/shim/service.go:158 +0xb8
github.com/containerd/containerd/linux/shim.(*client).State(0xc42010c310, 0x7f3d7b2a4248, 0xc4204706c0, 0x13aaf90, 0x0, 0x0, 0x0, 0xc4204a99c0, 0x3, 0xc4204a98b0)
	/home/sjd/go/src/github.com/containerd/containerd/linux/shim/client.go:53 +0x4c
github.com/containerd/containerd/linux.(*Container).State(0xc420109900, 0x7f3d7b2a4248, 0xc4204706c0, 0xc4201051e0, 0xc42001a500, 0xc4201257a8, 0xc42012bb00)
	/home/sjd/go/src/github.com/containerd/containerd/linux/container.go:53 +0x84
github.com/containerd/containerd/services/execution.containerFromContainerd(0x7f3d7b2a4248, 0xc4204706c0, 0x13669e0, 0xc420109900, 0xde5c80, 0x0, 0x1366c20)
	/home/sjd/go/src/github.com/containerd/containerd/services/execution/service.go:141 +0x63
github.com/containerd/containerd/services/execution.(*Service).List(0xc420109620, 0x7f3d7b2a4248, 0xc4204706c0, 0x13aaf90, 0x0, 0x0, 0x0)
	/home/sjd/go/src/github.com/containerd/containerd/services/execution/service.go:179 +0x152
github.com/containerd/containerd/api/services/execution._ContainerService_List_Handler.func1(0x7f3d7b2a4248, 0xc4204706c0, 0xdd7420, 0x13aaf90, 0xc42021c140, 0xc420108800, 0x14, 0xc4204a9a68)
	/home/sjd/go/src/github.com/containerd/containerd/api/services/execution/execution.pb.go:454 +0x86
github.com/containerd/containerd/vendor/github.com/grpc-ecosystem/go-grpc-prometheus.UnaryServerInterceptor(0x7f3d7b2a4248, 0xc4204706c0, 0xdd7420, 0x13aaf90, 0xc420108760, 0xc4201087c0, 0xc420234100, 0xc4201064a0, 0xc4204a9ae8, 0x771548)
	/home/sjd/go/src/github.com/containerd/containerd/vendor/github.com/grpc-ecosystem/go-grpc-prometheus/server.go:29 +0xa4
main.interceptor(0x7f3d7b2a4248, 0xc4204705a0, 0xdd7420, 0x13aaf90, 0xc420108760, 0xc4201087c0, 0xc42021e170, 0xc4204a9b60, 0x771548, 0x60)
	/home/sjd/go/src/github.com/containerd/containerd/cmd/containerd/main.go:425 +0x182
github.com/containerd/containerd/api/services/execution._ContainerService_List_Handler(0xdf3820, 0xc420109620, 0x7f3d7b2a4248, 0xc420470390, 0xc42030aae0, 0xe5d5a8, 0x0, 0x0, 0x0, 0x0)
	/home/sjd/go/src/github.com/containerd/containerd/api/services/execution/execution.pb.go:456 +0x177
github.com/containerd/containerd/vendor/google.golang.org/grpc.(*Server).processUnaryRPC(0xc4202340c0, 0x1366d40, 0xc42011a000, 0xc420206200, 0xc42024e210, 0x1351be0, 0xc4204704b0, 0x0, 0x0)
	/home/sjd/go/src/github.com/containerd/containerd/vendor/google.golang.org/grpc/server.go:708 +0xa61
github.com/containerd/containerd/vendor/google.golang.org/grpc.(*Server).handleStream(0xc4202340c0, 0x1366d40, 0xc42011a000, 0xc420206200, 0xc4204704b0)
	/home/sjd/go/src/github.com/containerd/containerd/vendor/google.golang.org/grpc/server.go:892 +0x1261
github.com/containerd/containerd/vendor/google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc420106170, 0xc4202340c0, 0x1366d40, 0xc42011a000, 0xc420206200)
	/home/sjd/go/src/github.com/containerd/containerd/vendor/google.golang.org/grpc/server.go:468 +0xa9
created by github.com/containerd/containerd/vendor/google.golang.org/grpc.(*Server).serveStreams.func1
	/home/sjd/go/src/github.com/containerd/containerd/vendor/google.golang.org/grpc/server.go:469 +0xa1
```

The running instance was built with 2369f498f2b1f5974bef0ee878ef89b6fbc1941e.

--- a/linux/shim.go
+++ b/linux/shim.go
@@ -24,7 +24,7 @@ import (
 
 func newShim(path string, remote bool) (shim.ShimClient, error) {
 	if !remote {
-		return localShim.Client(path), nil
+		return localShim.Client(path)
 	}
 	socket := filepath.Join(path, "shim.sock")
 	l, err := sys.CreateUnixSocket(socket)
@@ -59,7 +59,7 @@ func newShim(path string, remote bool) (shim.ShimClient, error) {
 
 func loadShim(path string, remote bool) (shim.ShimClient, error) {
 	if !remote {
-		return localShim.Client(path), nil
+		return localShim.Client(path)
 	}
 	socket := filepath.Join(path, "shim.sock")
 	return connectShim(socket)

--- a/linux/shim/client.go
+++ b/linux/shim/client.go
@@ -5,6 +5,7 @@ import (
 
 	shimapi "github.com/containerd/containerd/api/services/shim"
 	"github.com/containerd/containerd/api/types/container"
+	runc "github.com/crosbymichael/go-runc"
 	google_protobuf "github.com/golang/protobuf/ptypes/empty"
 	"golang.org/x/net/context"
 	"golang.org/x/sys/unix"
@@ -12,10 +13,26 @@ import (
 	"google.golang.org/grpc/metadata"
 )
 
-func Client(path string) shimapi.ShimClient {
-	return &client{
+func Client(path string) (shimapi.ShimClient, error) {
+	pid, err := runc.ReadPidFile(filepath.Join(path, "init.pid"))
+	if err != nil {
+		return nil, err
+	}
+
+	cl := &client{
 		s: New(path),
 	}
+
+	// used when quering  container status and info
+	cl.s.initProcess = &initProcess{
+		id:  filepath.Base(path),
+		pid: pid,
+		runc: &runc.Runc{
+			Log:       filepath.Join(path, "log.json"),
+			LogFormat: runc.JSON,
+		},
+	}
+	return cl, nil
 }
 
 type client struct {

